silq/tools/plot_tools.py

```diff
@@ -821,7 +833,15 @@ def update_plot(self, initialize=False):
             initialize: Method called during initialization.
         """
         for k, name in enumerate(self.parameter.names):
-            result = self.parameter.results[name]
+            if self.buffer_length > 1:
+                assert self.parameter.trace_pulse.enabled is False, \
+                    'Trace pulse must be turned off for running average.'
+                self.buffers[self.buf_idx] = self.parameter.results[name]
+                self.buf_idx = (self.buf_idx + 1) % self.buffer_length
+                result = np.mean([buf for buf in self.buffers if buf is not None], axis=0)
```

Comment:
Does this still return a measurement every acquisition, or only once every `buffer_length` acquisitions? It looks like it's returning every acquisition

Reply:
yes, every acquisition - the idea is to see yourself how the data quality improves with time the longer you wait